lib/core/src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -471,6 +473,9 @@ isSubrangeOf r1 r2 =
                                   Stake Pools
 -------------------------------------------------------------------------------}
 
+type Stake = Quantity "lovelace" Word64
+type EpochNo = Word64
```

Comment:
No :no_mouth: ... Don't use type aliases here. Those types are simple enough, if they deserve to exist as types, make them newtypes. Though, `Stake`, has no extra semantic compared to `Quantity "lovelace" Word64`

Reply:
I find the `Quantity "lovelace" Word64` a bit clunky and to contain be mostly un-interesting information.

Readability and formatting gets more difficult. For example this fits into one line with the alias
```
        => WhenMissing (ExceptT ErrListStakePools m) PoolId Stake (Stake, Int)

```
but not without:
```
    stakeButNoActivity
        :: Monad m
        => WhenMissing
            (ExceptT ErrListStakePools m)
            PoolId
            (Quantity "lovelace" Word64)
            (Quantity "lovelace" Word64, Int)
```
😥 

I wonder if we can remove some `Quantity "lovelace"` in favour of `Lovelace deriving X via Quantity "lovelace"` eventually…

Reply:
The `Quantity` is really just to avoid carrying around naked types like `Int`, or `Word32` and instead, ease type signatures readability. We also use it extensively in the API to disambiguate JSON representation of quantified units. 

Yet, sometimes it makes sense to define a newtype if one really wants to capture a particular meaning (for examples: https://github.com/input-output-hk/cardano-wallet/blob/master/lib/core/src/Cardano/Wallet/Primitive/Types.hs#L1179-L1203). A `Stake` type could work but then, it becomes semantically awkward to me as we are basically saying "this is not any lovelace amount, it's a stake so it's kinda special", whereas, it's not. But, if you think having a dedicated type for this makes more sense, okay. The `Quantity` is really just there so that we don't have to create newtype all the time when what we just want to clarify the meaning given to some raw values (precisely like `Hash` for raw `ByteString`).

Having said that... The argument  "I can't indent it in one line" is pretty weak to justify a new type :grimacing: